lib/livebook/apps.ex

```diff
@@ -186,7 +188,12 @@ defmodule Livebook.Apps do
 
       if Livebook.Notebook.AppSettings.valid?(notebook.app_settings) do
         warnings = Enum.map(warnings, &("Import: " <> &1))
-        deploy(notebook, warnings: warnings)
+
+        case Livebook.Hubs.get_offline_hub() do
```

Comment:
@josevalim do we want this behaviour to be offline-hub specific, or should there be a separate env var with the restricted hub name/id, so it works with online hubs too?

Reply:
Wouldn't it be the idea of the offline hub? To be able to deploy apps using docker with the offline hub set up by env vars.

Reply:
There are two features, which in theory are orthogonal:

* being able to deploy with teams hub offline (which as opposed to personal hub enables certain features, involves encrypting secrets with the key, etc)

* being able to restrict deployment to verified notebooks from a specific hub (which could be online)

Perhaps we want to keep the behaviour tied together, hence my question :)

Reply:
We don't have to couple it. We could add a `LIVEBOOK_APPS_PATH_ENFORCE_HUB=...` but then it means one additional configuration. So the trade-off is between fewer configuration and more control. Can we think of another use case where this feature would be desired?

Reply:
Ah it is specific to `LIVEBOOK_APPS_PATH`, which is headless and rather won't be used with online teams hub, fair. So the second question is whether it should be opt-in for offline hub, but we can also add that later if there is a use case.

Reply:
We should document the restrictive behaviour when `LIVEBOOK_TEAMS_*` are set, perhaps under `LIVEBOOK_APPS_PATH`.

Reply:
I guess someone could deploy a Livebook with apps path and still run on online mode for additional apps. I think the biggest issue about coupling this behaviour is that we cannot change it without backwards incompatible changes, unless the option we introduce it is about "not doing the thing we do by default when offline hubs are used". So I retract my statement I think we should call this `LIVEBOOK_APPS_PATH_HUB_ID`. It may make testing easier too.